### PR TITLE
AutoQuantize minor improvement: limit grad enabled parameters, limit cpu-gpu sync during scoring

### PR DESCRIPTION
**Pull request by realAsma**
## What does this PR do?

**Type of change:** ? 
AutoQuantize minor improvement: limit grad enabled parameters, limit cpu-gpu sync during scoring

**Overview:** ?

Minor improvements for AutoQuantize.
Added support to enable grad only for selected parameters - this should reduce the number of gemms In the backward pass by 1x.

## Testing
Covered by unit tests (both CPU and GPU)

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes
- **Did you write any new necessary tests?**: Yes
- **Did you add or update any necessary documentation?**: NA
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: NA

### CODE REVIEW

File: modelopt/torch/quantization/algorithms.py [hunk 1]
```diff
@@ -238,11 +242,15 @@ def active(self, val: HPType | None):
     def importance(self) -> dict:
         """Return the importance dict mapping recipe and importance."""
         return {
-            quant_recipe: sum(importance_dict.values())
+            quant_recipe: sum(v.cpu().item() for v in importance_dict.values())
```
Comment (cjluo-nv) [inline modelopt/torch/quantization/algorithms.py:245]: why do we need to move it to the cpu here?

File: modelopt/torch/quantization/algorithms.py [hunk 2]
```diff
@@ -261,7 +269,7 @@ class AutoQuantizeSearcher(BaseSearcher):
 
     candidate_stats: dict[str, dict[str, list[float]]]
     best: dict[str, Any]
-    gradient_checkpointing_enable_contexts: list[tuple[Callable, Callable]] = []
+    custom_support: list[tuple[Callable, Callable, Callable]] = []
```
Comment (cjluo-nv) [inline modelopt/torch/quantization/algorithms.py:272]: could you document the class members?

File: modelopt/torch/quantization/plugins/huggingface.py [hunk 3]
```diff
@@ -739,8 +739,14 @@ def setup_model_for_gradient_checkpointing(model: nn.Module):
         model.config.use_cache = use_cache
 
 
-AutoQuantizeSearcher.register_gradient_checkpointing_enable_context(
-    _is_supported_hf_model, setup_model_for_gradient_checkpointing
+def _is_param_grad_enabled_for_auto_quantize(pname, model):
+    return "embed" in pname
```
Comment (cjluo-nv) [inline modelopt/torch/quantization/plugins/huggingface.py:745]: does it mean we only enable gradient for the "embed" related parameters?

File: modelopt/torch/quantization/algorithms.py (hunk 1 above)
Comment (realAsma) [reply modelopt/torch/quantization/algorithms.py:245]: the search is performed on cpu. the score is on GPU.

File: modelopt/torch/quantization/plugins/huggingface.py (hunk 3 above)
Comment (realAsma) [reply modelopt/torch/quantization/plugins/huggingface.py:745]: yes, that should be enough to trigger autograd computation for transformer linear layers

File: modelopt/torch/quantization/algorithms.py (hunk 2 above)
Comment (realAsma) [reply modelopt/torch/quantization/algorithms.py:272]: I am working a few more other PR on auto_quantize improvements. I will improve the doc in the final PR. Does that work?

Comment (cjluo-nv) [reply modelopt/torch/quantization/algorithms.py:272]: I feel it would be great to have a PR self-contained. If this does not cost too much engineering bandwidth, how about just do it right in this PR?

File: modelopt/torch/quantization/plugins/huggingface.py (hunk 3 above)
Comment (cjluo-nv) [reply modelopt/torch/quantization/plugins/huggingface.py:745]: Could you elaborate (And maybe document in the code)?

File: modelopt/torch/quantization/algorithms.py (hunk 2 above)
Comment (realAsma) [reply modelopt/torch/quantization/algorithms.py:272]: I am making more auto_quantize changes in the upcoming PRs. I would need to do double work since I would need to change the doc I add in this PR in the upcoming PRs

Comment (realAsma) [reply modelopt/torch/quantization/algorithms.py:272]: In addition, the above is not a user facing interface - it is a framework facing interface - such as an interface for huggingface, interface for Megatron etc.

File: modelopt/torch/quantization/plugins/huggingface.py (hunk 3 above)
Comment (realAsma) [reply modelopt/torch/quantization/plugins/huggingface.py:745]: Done.

Comment (realAsma) [reply modelopt/torch/quantization/plugins/huggingface.py:745]: We enable gradients only for embedding layers to ensure gradient flow throughout the model during AutoQuantize's sensitivity estimation, while avoiding the memory cost of computing and storing gradients for millions of other parameters.
